src/dao/AbstractContractDAO.js

```diff
@@ -378,8 +386,10 @@ export default class AbstractContractDAO {
           throw new TxError('Dry run failed', dryResult)
         }
 
+        // TODO @dkchv: here, see above
+
         /** TRANSACTION */
-        await AbstractContractDAO.txStart(tx)
+        await AbstractContractDAO.txStart(tx, plural)
```

Comment:
if tx is plural, then we need to show our custom confirmation modal for every provider, even for metamask/mist users, so they will able to see our message about plurality